src/World.h

```diff
@@ -267,13 +267,13 @@ class cWorld :
 	Uses a queue to store the player object until the Tick thread processes the addition event.
 	Also adds the player as an entity in the chunkmap, and the player's ClientHandle, if any, for ticking.
 	If a_OldWorld is provided, a corresponding ENTITY_CHANGED_WORLD event is triggerred after the addition. */
-	void AddPlayer(cPlayer * a_Player, cWorld * a_OldWorld = nullptr);
+	void AddPlayer(std::unique_ptr<cPlayer> a_Player, cWorld * a_OldWorld = nullptr);
 
 	/** Removes the player from the world.
 	Removes the player from the addition queue, too, if appropriate.
 	If the player has a ClientHandle, the ClientHandle is removed from all chunks in the world and will not be ticked by this world anymore.
 	@param a_RemoveFromChunk determines if the entity should be removed from its chunk as well. Should be false when ticking from cChunk. */
-	void RemovePlayer(cPlayer * a_Player, bool a_RemoveFromChunk);
+	std::unique_ptr<cPlayer> RemovePlayer(cPlayer & a_Player, bool a_RemoveFromChunk);
```

Comment:
An aside; did we decide to use Javadoc-style comments (the `@param` tag)? I believe XML comments are fully supported by Doxygen.